internal/bminventory/inventory.go

```diff
@@ -1315,6 +1317,25 @@ func (b *bareMetalInventory) TransformClusterToDay2Internal(ctx context.Context,
 		return nil, common.NewApiError(http.StatusInternalServerError, err)
 	}
 
+	if cluster.IgnitionEndpoint == nil || cluster.IgnitionEndpoint.URL == nil {
+		// Set custom Ignition endpoint so that day2 workers would join using HTTPS endpoint
+		// ensureMCSCert would add ignition override with MCS secret
+		ignitionEndpoint := fmt.Sprintf("%s/config/%s", common.GetMCSUrlBase(cluster), models.HostRoleWorker)
```

Comment:
I know I'm reviewing a WIP, but you have just hardcoded `HostRoleWorker` in a function that from its name does not seem to be dedicated purely for workers